internal/pkg/service/buffer/store/watcher_test.go

```diff
@@ -322,3 +323,43 @@ func TestStore_Watcher_HandleExportEvent(t *testing.T) {
 	_, found = mappings.(map[key.ExportKey]*model.Mapping)[expKey]
 	assert.False(t, found)
 }
+
+func TestStore_Watcher_Watch(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	store := newStoreForTest(t)
+
+	// Init watcher
+	w, err := NewWatcher(ctx, store)
+	assert.NoError(t, err)
+	w.Watch(ctx, log.NewNopLogger(), store.client)
+
+	export := receiver.Exports[0]
+	fileID, _ := time.Parse(time.RFC3339, "2006-01-01T15:04:05+07:00")
+	fileID = fileID.UTC()
+	file := model.File{
+		FileKey: key.FileKey{
+			ExportKey: export.ExportKey,
+			FileID:    fileID,
+		},
+		Mapping:         export.Mapping,
+		StorageResource: &storageapi.File{ID: 123},
+	}
+	sliceID := fileID.Add(time.Hour)
+	slice := model.Slice{
+		SliceKey: key.SliceKey{
+			FileKey: file.FileKey,
+			SliceID: sliceID,
+		},
+		SliceNumber: 1,
+	}
+
+	// New slice adds to w.slicesForExports
+	_, found := w.slicesForExports.Load(export.ExportKey)
+	assert.False(t, found)
+	_, err = store.createSliceOp(ctx, slice).Do(ctx, store.client)
+	assert.NoError(t, err)
+	_, found = w.slicesForExports.Load(export.ExportKey)
+	assert.True(t, found)
```

Comment:
@michaljurecko Vypadá to tak, že ten watcher nejede. 😕 A to už ani sem se v testu zřejmě nedostane: https://github.com/keboola/keboola-as-code/blob/33203eaf0f9063c97d8df8b2c70c9f7c19f921ac/internal/pkg/service/common/etcdop/prefix.go#L158-L159

A nefunguje mi to teda ani takto:
![image](https://user-images.githubusercontent.com/215660/206503990-b18f2415-ae67-414f-9577-56079234fb40.png)
To se nemusí nějak extra zapínat, ne?

 (Pokoušel jsem se to teda debugovat přes IntelliJ, ale s tím etcd to asi nejde bez docker-compose a to se mi nedaří rozjet, tak jsem odkázanej na přidávání debug logů do kódu. 😅 )

Reply:
pozriem to rano podrobnejsie, ... ale chyba mi tam nejaky sleep /  (lepsie wait group) na konci, ktora by cakala, ze sa da debugovacia goroutine nejak prejavi